blocktx/peer_handler.go

```diff
@@ -365,38 +353,32 @@ func (bs *PeerHandler) insertBlock(blockHash *chainhash.Hash, merkleRoot *chainh
 	return bs.store.InsertBlock(context.Background(), block)
 }
 
-func (bs *PeerHandler) markTransactionsAsMined(blockId uint64, transactionHashes []*chainhash.Hash, merkleTree []string) error {
+func (bs *PeerHandler) markTransactionsAsMined(blockId uint64, merkleTree []*chainhash.Hash, blockHeight uint64) error {
 	start := gocore.CurrentNanos()
 	defer func() {
 		gocore.NewStat("blocktx").NewStat("HandleBlock").NewStat("markTransactionsAsMined").AddTime(start)
 	}()
 
 	txs := make([]*blocktx_api.TransactionAndSource, 0, TransactionStoringInterval)
 	merklePaths := make([]string, 0, TransactionStoringInterval)
+	transactionHashes := merkleTree[:(len(merkleTree)+1)/2]
```

Comment:
Do we really have that many transactions in the block? What if number of transactions is not power of 2 ? 🤔  Correct me if I'm wrong but will this line be correct ?

Reply:
Resolved, called them leaves.